src/charm.py

```diff
@@ -96,18 +94,60 @@ def _are_relations_ready(self) -> bool:
         return self._tls.get_tls_relation() is not None
 
     def _reconcile(self) -> None:
-        """Reconcile charm status based on configuration and integrations."""
+        """Reconcile charm status based on configuration and integrations.
+
+        Raises:
+            RuntimeError: when initializing the lightkube client fails,
+            or when creating the gateway resource fails.
+        """
+        try:
+            # Set field_manager for server-side apply when patching resources
+            # Keep this consistent across client initializations
+            kubeconfig = KubeConfig.from_service_account()
+            client = Client(config=kubeconfig, field_manager=self.app.name)
+        except ConfigError as exc:
+            logger.error("Error initializing the lightkube client: %s", exc)
+            raise RuntimeError("Error initializing the lightkube client.") from exc
```

Comment:
I think `logger.exception` might be enough here to print the exception stack trace :D
https://docs.python.org/3/library/logging.html#logging.Logger.exception

Reply:
And other `logger.error`s from exceptions :)

Reply:
got it, I updated from `logger.error` to `logger.exception` in relevant places.